framework/include/actions/Action.h

```diff
@@ -44,8 +46,48 @@ class Action : public ConsoleStreamInterface
 
   virtual void act() = 0;
 
+  /**
+   * The name of the action
+   *
+   * When using MOOSE input files, this is the last portion
+   * of the full input syntax:
+   *    [System]
+   *      [./name]
+   *
+   *    [System]
+   *      [./Subsystem]
+   *        [./name]
+   */
   const std::string & name() const { return _name; }
 
+  ///@{
+  /**
+   * The full input file and base syntax for the action
+   *
+   * _full_syntax = System/name
+   * _base_syntax = System
+   * [System]
+   *   [./name]
+   *
+   * _full_syntax = System/Subsystem/name
+   * _base_syntax = System/Subsystem
+   * [System]
+   *   [./Subsystem]
+   *     [./name]
+   */
+  const std::string & getInputSyntax() const { return _full_syntax; }
```

Comment:
This is nothing new, I am just calling it what it is. Before this was the "long_name". 

The only reason that this is there is because in some app actions the "extractParams" method is called to extract syntax up the hierarchy of syntax, this will be removed if we implement the parameter traversing stuff.